### PR TITLE
8318455: Fix the compiler/sharedstubs/SharedTrampolineTest.java and SharedStubToInterpTest.java

### PR DESCRIPTION
**Pull request by lepestock**
A continuation of #16258.

Said tests try to control the compilers used via arguments passed to subprocesses' they create. The [JDK-8314838](https://bugs.openjdk.org/browse/JDK-8314838) added support for additional forwarding of the JTReg javaoptions to the subprocesses, but broke the original arguments passing. They are now passed to the enclosing driver and simply ignored by ProcessTools.createTestJvm. As a result, proper compiler may be not chosen on subprocess start, etc. This PR fixes that.

Testing: both tests run in CI, with additional arguments (tests were ignored, as expected) and without (passed).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8318455](https://bugs.openjdk.org/browse/JDK-8318455): Fix the compiler/sharedstubs/SharedTrampolineTest.java and SharedStubToInterpTest.java (**Bug** - P4)

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)
 * @yftsai (no known openjdk.org user name / role)
 * [Evgeny Astigeevich](https://openjdk.org/census#eastigeevich) (@eastig - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16277/head:pull/16277` \
`$ git checkout pull/16277`

Update a local copy of the PR: \
`$ git checkout pull/16277` \
`$ git pull https://git.openjdk.org/jdk.git pull/16277/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16277`

View PR using the GUI difftool: \
`$ git pr show -t 16277`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16277.diff">https://git.openjdk.org/jdk/pull/16277.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16277#issuecomment-1771659959)

### CODE REVIEW

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

File: test/hotspot/jtreg/compiler/sharedstubs/SharedTrampolineTest.java [hunk 1]
```diff
@@ -49,6 +50,7 @@ public class SharedTrampolineTest {
     private static void runTest(String test) throws Exception {
         String testClassName = SharedTrampolineTest.class.getName() + "$" + test;
         ArrayList<String> command = new ArrayList<String>();
+        command.add("-XX:-TieredCompilation");  // C2-compiler only
```
Comment (shipilev) [inline test/hotspot/jtreg/compiler/sharedstubs/SharedTrampolineTest.java:53]: It is not quite clear to me why we should not pass `args[0]` here, like we do in another test fix. 
Would this test ever make sense to run with different compiler, e.g. C1?

Comment (lepestock) [reply test/hotspot/jtreg/compiler/sharedstubs/SharedTrampolineTest.java:53]: Superficially, it looks like the test is optimised for C2. But let's ask @eastig for help.

Comment (eastig) [reply test/hotspot/jtreg/compiler/sharedstubs/SharedTrampolineTest.java:53]: I agree with Aleksey we should use `args[0]`. It looks like we will need to add a test for C1.

Comment (lepestock) [reply test/hotspot/jtreg/compiler/sharedstubs/SharedTrampolineTest.java]: > I agree with Aleksey we should use args[0]. It looks like we will need to add a test for C1.

Got it, thank you. Fixed.

Comment (shipilev) [reply test/hotspot/jtreg/compiler/sharedstubs/SharedTrampolineTest.java]: Are we adding C1 version in this PR as well, or in a separate one?

Comment (eastig) [reply test/hotspot/jtreg/compiler/sharedstubs/SharedTrampolineTest.java]: I think it is better to add a new test with a separate PR.

**Review by shipilev — APPROVED** [review]
All right then! Looks fine.

**Review by yftsai — APPROVED** [review]

File: test/hotspot/jtreg/compiler/sharedstubs/SharedTrampolineTest.java (hunk 1 above)
Comment (eastig) [reply test/hotspot/jtreg/compiler/sharedstubs/SharedTrampolineTest.java]: Created https://bugs.openjdk.org/browse/JDK-8318681 to add tests for C1.

**Review by eastig — APPROVED** [review]
lgtm

**Review by TobiHartmann — APPROVED** [review]

**Pull request closed without merging**